src/vma/dev/allocator.cpp

```diff
@@ -90,20 +90,20 @@ void* vma_allocator::alloc_and_reg_mr(size_t size, ib_ctx_handler *p_ib_ctx_h)
 {
 	uint64_t access = VMA_IBV_ACCESS_LOCAL_WRITE;
 
-	switch (safe_mce_sys().mem_alloc_type) {
+	switch (m_mem_alloc_type) {
 	case ALLOC_TYPE_HUGEPAGES:
 		if (!hugetlb_alloc(size)) {
 			__log_info_dbg("Failed allocating huge pages, "
-				       "falling back to contiguous pages");
+				       "falling back to another memory allocation method");
 		}
 		else {
 			__log_info_dbg("Huge pages allocation passed successfully");
-			m_mem_alloc_type = ALLOC_TYPE_HUGEPAGES;
 			if (!register_memory(size, p_ib_ctx_h, access)) {
 				__log_info_dbg("failed registering huge pages data memory block");
 				throw_vma_exception("failed registering huge pages data memory"
 						" block");
 			}
+			m_mem_alloc_type = ALLOC_TYPE_HUGEPAGES;
```

Comment:
I agree with rafi, also this change should be reverted.

Reply:
I think that setting `m_mem_alloc_type = ALLOC_TYPE_ANON;` should be done after we see that allocation/registration are done successfully.
According current implementation VMA catches exception in case failure during register_memory().